stdlib/src/testing/testing.mojo

```diff
@@ -41,21 +41,32 @@ fn _isclose(
     rtol: Scalar[a.type],
     equal_nan: Bool,
 ) -> SIMD[DType.bool, a.size]:
+    constrained[
+        a.type.is_bool() or a.type.is_integral() or a.type.is_floating_point(),
+        "input type must be boolean, integral, or floating-point",
+    ]()
+
     @parameter
     if a.type.is_bool() or a.type.is_integral():
         return a == b
+    else:
+        var both_nan = (isnan(a) & isnan(b))
+        if equal_nan and both_nan.reduce_and():
+            return True
+
+        var res = (a == b)
+        var atol_vec = SIMD[a.type, a.size](atol)
+        var rtol_vec = SIMD[a.type, a.size](rtol)
+        res |= (
+            isfinite(a)
+            & isfinite(b)
+            & (_abs(a - b) <= (atol_vec.max(rtol_vec * _abs(a).max(_abs(b)))))
+        )
 
-    if equal_nan and isnan(a) and isnan(b):
-        return True
-
-    var atol_vec = SIMD[a.type, a.size](atol)
-    var rtol_vec = SIMD[a.type, a.size](rtol)
-    var res = _abs(a - b) <= (atol_vec.max(rtol_vec * _abs(a).max(_abs(b))))
-
-    if not equal_nan:
-        return res
+        if not equal_nan:
```

Comment:
I think ternary is a bit more clearer here, and we can avoid the negated form `not equal_nan`.